src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -1869,7 +1809,7 @@ public virtual bool Accepts(DType type, out KeyValuePair<string, DType> schemaDi
                 case DKind.Record:
                 case DKind.File:
                 case DKind.LargeImage:
-                    if (Kind == type.Kind)
+                    if (Kind == type.Kind || type.IsLazyType)
```

Comment:
This ensures that `DType.EmptyRecord.Accepts(LazyRecord)`, so all our functions that operate on arbitrary table/record types behave correctly. 
We may also want to add support for `SomeSpecificRecord.Accepts(LazyRecord)`, however, that's not currently supported for Relationships/Controls/... and so that's left out until there is demand for it.